rayon-core/src/registry.rs

```diff
@@ -354,6 +356,17 @@ impl Registry {
         let job = StackJob::new(|injected| {
             let worker_thread = WorkerThread::current();
             assert!(injected && !worker_thread.is_null());
+            let worker_thread = &*worker_thread;
+            worker_thread.registry.sleep.add_external_waiter();
```

Comment:
I don't really understand what an "external waiter" is

Reply:
Oh, I ... guess I maybe do. It's just some external thread that is blocked waiting on Rayon to finish up?

Reply:
What is the reason to have this? Like, wouldn't we want to be called back regardless if the worker threads have deadlocked?

Reply:
As I wrote on IRC, I am wondering now if we can remove the concept of "external waiter", and instead just only issue a callback when there is at least one blocked thread.

Reply:
I've removed the external waiter counter.